itests/worker_test.go

```diff
@@ -401,3 +404,75 @@ func TestWindowPostWorkerManualPoSt(t *testing.T) {
 	require.NoError(t, err)
 	require.Len(t, lastPending, 0)
 }
+
+func TestSchedulerRemoveRequest(t *testing.T) {
+	ctx := context.Background()
+	_, miner, worker, ens := kit.EnsembleWorker(t, kit.WithAllSubsystems(), kit.ThroughRPC(), kit.WithNoLocalSealing(true),
+		kit.WithTaskTypes([]sealtasks.TaskType{sealtasks.TTFetch, sealtasks.TTCommit1, sealtasks.TTFinalize, sealtasks.TTDataCid, sealtasks.TTAddPiece, sealtasks.TTPreCommit1, sealtasks.TTCommit2, sealtasks.TTUnseal})) // no mock proofs
+
+	ens.InterconnectAll().BeginMining(50 * time.Millisecond)
+
+	e, err := worker.Enabled(ctx)
+	require.NoError(t, err)
+	require.True(t, e)
+
+	type SchedDiagRequestInfo struct {
+		Sector   abi.SectorID
+		TaskType sealtasks.TaskType
+		Priority int
+		SchedId  uuid.UUID
+	}
+	type SchedDiagInfo struct {
+		Requests    []SchedDiagRequestInfo
+		OpenWindows []string
+	}
+	type i struct {
+		SchedInfo    interface{}
+		ReturnedWork []string
+		Waiting      []string
+		CallToWork   map[string]string
+		EarlyRet     []string
+	}
+
+	go miner.PledgeSectors(ctx, 1, 0, nil)
+	// Keep checking till sector state is PC2, the request should get stuck as worker cannot process PC2
+	for {
+		st, err := miner.SectorsStatus(ctx, abi.SectorNumber(1), false)
+		require.NoError(t, err)
+		if st.State == api.SectorState(sealing.PreCommit2) {
+			break
+		}
```

Comment:
Probably want to add some sleep here
```suggestion
		}
		time.Sleep(time.Second)
```